specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1409 @@
+    /// @param minVerrsion the minimum version to pin
+    /// @param options additional options affecting the package dependency
+    ///
+    /// @note This fails if the package dependency cannot be resolved to a specific package.
+    ///
+    /// @see Pin(String, PackageVersion)
+    /// @see Pin(String, PackageVersion, PinPackageDependencyOptions)
+    /// @see PinForSystem()
+    static PackageDependency PinForUser(
+        Windows.System.User user,
+        String packageFamilyName,
+        PackageVersion minVersion,
+        PinPackageDependencyOptions options);
+
+    /// Define a package dependency for the system (i.e. all users). The criteria for a PackageDependency
+    /// (package family name, minimum version, etc) may match multiple
+    /// packages, but ensures Deployment won't remove a package if it's
+    /// the only one satisfying the PackageDependency.
+    ///
+    /// @note A package matching a PackageDependency pin can still be removed
+    ///       as long as there's another that satisfies the PackageDependency.
+    ///       For example, if Fwk-v1 is installed and a PackageDependency specifies
+    ///       MinVersion=1 and then Fwk-v2 is installed, Deployment could remove
+    ///       Fwk-v1 because Fwk-v2 will satisfy the PackageDependency. After Fwk-v1
+    ///       is removed Deployment won't remove Fwk-v2 because it's the only package
+    ///       satisfying the PackageDependency. Thus  Fwk-v1 and Fwk-v2 (and any other
+    ///       package matching the PackageDependency) are 'loosely pinned' â€“ Deployment
+    ///       guarantees it won't remove a package if it would make a PackageDependency
+    ///       unsatisfied.
+    ///
+    /// A PackageDependency specifies criteria (package family, minimum version, etc)
+    /// and not a specific package. Deployment reserves the right to use a different
+    /// package (e.g. higher version) to satisfy the PackageDependency if/when
+    /// one becomes available.
+    ///
+    /// @param user the user scope of the package dependency. If null the caller's user context is used
+    /// @param packageFamilyName the package family to pin
+    /// @param minVerrsion the minimum version to pin
+    /// @param options additional options affecting the package dependency
+    ///
+    /// @note This fails if the package dependency cannot be resolved to a specific package.
+    ///       This package resolution check is skipped if MddPinPackageDependency.DoNotVerifyDependencyResolution
+    ///       is specified. This is useful for installers pinning a package dependency for all users on a system.
+    ///
+    /// @see Pin(String, PackageVersion)
+    /// @see Pin(String, PackageVersion, PinPackageDependencyOptions)
+    /// @see PinForUser()
+    static PackageDependency PinForSystem(
+        String packageFamilyName,
+        PackageVersion minVersion,
+        PinPackageDependencyOptions options);
+
+    void Unpin();
```

Comment:
Is the ID valid after unpinning?

Reply:
DISCUSSION: No, the ID string is useless after unpin/delete. The property is valid, but attempting to `PackageDependency.GetFromId` returns null.